src/crypto-api.ts

```diff
@@ -372,10 +372,13 @@ export interface CryptoApi {
      * Store the backup decryption key.
      *
      * This should be called if the client has received the key from another device via secret sharing (gossiping).
+     * It is the responsability of the caller to check that the decryption key is valid to the given backup version.
      *
      * @param key - the backup decryption key
+     * @param version - the backup version corresponding to this decryption key
+     *
      */
-    storeSessionBackupPrivateKey(key: Uint8Array): Promise<void>;
+    storeSessionBackupPrivateKey(key: Uint8Array, version: string): Promise<void>;
```

Comment:
this is a breaking change. We need to continue to expose the old API, even if it only works on legacy crypto.